core/src/main/java/org/apache/iceberg/view/ViewUtil.java

```diff
@@ -18,12 +18,34 @@
  */
 package org.apache.iceberg.view;
 
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.iceberg.Schema;
 import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.types.TypeUtil;
 
 public class ViewUtil {
   private ViewUtil() {}
 
   public static String fullViewName(String catalog, TableIdentifier ident) {
     return catalog + "." + ident;
   }
+
+  public static Schema assignFreshIds(ViewMetadata base, Schema newSchema) {
+    return TypeUtil.assignFreshIds(
+        newSchema,
+        base.schema(),
+        new AtomicInteger(highestFieldId(base.schemas()))::incrementAndGet);
+  }
+
+  public static int maxVersionId(ViewMetadata metadata) {
```

Comment:
This method has the same issue. Instead of passing `ViewMetadata`, it should accept `Collection<ViewVersion>`.

Reply:
I've removed this refactoring around `maxVersionId` for now since assigning fresh ids moved into `ViewMetadata` and it didn't seem appropriate anymore to include this refactoring (I'll open a separate PR for this)